htlcswitch/switch.go

```diff
@@ -524,7 +524,11 @@ func (s *Switch) GetAttemptResult(attemptID uint64, paymentHash lntypes.Hash,
 			return
 		}
 		resultChan <- result
-	}()
+	})
+	// The switch shutting down is signaled by closing the channel.
+	if !ok {
```

Comment:
Why do we still need this check? Won't the line `<-ctx.Done()` be hit when it's shutting down?

Reply:
If `GoroutineManager.Stop` is called before the `Go` method (i.e., the switch is in the process of stopping), the `Go` method will return `false` without launching a new goroutine. In such cases, we should perform the same action as if it had stopped after launching the goroutine - specifically, closing `resultChan`. Failing to close the channel and simply returning it could cause the caller to get stuck indefinitely while waiting to receive from the channel.